### PR TITLE
add unique identifier to messages

### PR DESCRIPTION
**Pull request by reggie3**
Addresses issue #11

### CODE REVIEW

File: WebView.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
-import EventEmitter from 'events';
-import React, { Component } from 'react';
-import { WebView as NativeWebView } from 'react-native';
+import EventEmitter from "events";
+import React, { Component } from "react";
+import { WebView as NativeWebView } from "react-native";
```
Comment (R1ZZU) [inline WebView.js]: I'd leave single quotes everywhere .This is common and well known part of javascript styleguide. Apart of that, every project has it's own styleguide as well all developers have there own coding style/preferences. If smth seems really weird to one, this should be raised as discussion point. Codestyle changes shouldn't be a part of feature pull request/bug fix

File: WebView.js [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
+
+// package unique prefix for all messages
+const UNIQUE_MESSAGE_PREFIX = "f251c210-e7c9-42fa-bae3-b9352ec3722a";
```
Comment (R1ZZU) [inline WebView.js]: not sure if this identifier should be SO unique 😄
One more point: this identifier should be shared between `react-native` and `WebView` so I'd better move it to separate file. Also `UNIQUE_MESSAGE_PREFIX` is a good name in general, but in that exact case I'd add more context to name to make it obvious that it relates to this package. `RN_MESSAGES_CHANNEL_PREFIX` works fine IMO

File: WebView.js [hunk 3]
```diff
@@ -18,42 +21,58 @@ export class WebView extends Component {
     );
   }
 
-  _refWebView = (webview) => {
```
Comment (R1ZZU) [inline WebView.js]: unjustified codestyle related changes (see first comment)

File: WebView.js [hunk 4]
```diff
@@ -18,42 +21,58 @@ export class WebView extends Component {
...
-  }
+  };
 
-  onMessage = (event) => {
+  onMessage = event => {
     const { data } = event.nativeEvent;
-    const parsedMsg = JSON.parse(data);
+
+    if (data.indexOf(UNIQUE_MESSAGE_PREFIX) !== 0) {
+      return; // that's not something that was received from rn messages channel
```
Comment (R1ZZU) [inline WebView.js]: this comment is redundant, as code is quite simple and self describing

**Comment by R1ZZU** [comment]
Just to summarize: I see a lot of codestyle realted changes, so I think I should add eslint to avoid redundant changes

**Comment by reggie3** [comment]
I think my cod prettifier mist have changed the singles to doubles. As fir  the unique ID, I figured better safe than sorry, and it'd give you the option to change it. I'll see about making the changes you mentioned.

**Comment by reggie3** [comment]
I'm not sure if my fixes for the code review issues you brought up made it through.  Resubmitting a pull seems to work differently then doing it the first time.  I changed the double quotes to single, removed the code style issues that I was able to figure out, and created a separate config file to contain the unique ID.  I didn't shorten the ID though since I felt you'd probably have something specific in mind.

**Comment by R1ZZU** [comment]
Now it looks really good! Thanks 👍 (will merge this one after verification everything works as expected)

**Comment by reggie3** [comment]
I have downloaded this package via NPM since this pull request was merged and the version that I receive does not appear to have received the update.  It doesn't include the config file or any use of it.

**Comment by R1ZZU** [comment]
I've just published `v1.2.0` to npm, this version should include this fix

**Comment by reggie3** [comment]
Thanks. I thought I was missing something.

**Pull request merged**